File: 1/p1-1.py
sum = 0


def handleLine(line: str) -> int:
    numbers = "0123456789"
    numsFound = []
    for c in line:
        if c in numbers:
            numsFound.append(c)

    return int(numsFound[0] + numsFound[-1])


with open('p1input.txt', 'r') as f:
    for line in f:
        sum += handleLine(line)

print(sum)

File: 1/p1-1.hs
import GHC.IO.IOMode (IOMode(ReadMode))
import Data.ByteString (hGetContents)
import Control.Exception (handle)
import System.IO (openFile)
import Data.ByteString as BS (putStr)
import Text.Read (readMaybe)
import Data.Char (digitToInt)
import GHC.Unicode (isDigit)

main = do
    let file = "p1input.txt"
    handle <- openFile file ReadMode
    contents <- hGetContents handle
    BS.putStr contents

handleFile :: [String] -> [String]
handleFile contents =
    [ x | x <- contents]

handleLine :: String -> String
handleLine = filter isDigit
